### PR TITLE
refactor: upgrade to vuepress 2

### LINKED ISSUE #190: Fix security issues
From https://github.com/localgovdrupal/docs/security/dependabot

image

### LINKED ISSUE #194: Site title/logo is obscured on some screen widths
image

Seems to happen at viewport widths between 1220-1480px

### PR DESCRIPTION
**Pull request by millnut**
## What does this change?

- Update Vuepress from v1 to v2
- Update Node from 14 to 20 LTS, 14 is no longer supported
- Update all core plugins and themes
- Update to support new theming
- Update child theme handling and to match the brand
- Add `.nvmrc` to let netlify know which Node version we require
- Update images path as image aliases are no longer available
- Update search config
- Fixes #190
- Fixes #194 Update the navbar items so the search bar is visible between 1200px and 1350px

### CODE REVIEW

**Comment by millnut** [comment]
Hi @stephen-cox this is still in progress while I look at Algolia integration however the preview fails due to using Node 16.

```
error @vitejs/plugin-vue@5.1.2: The engine "node" is incompatible with this module. Expected version "^18.0.0 || >=20.0.0". Got "16.20.2"
```

Can the node version for preview environments be updated to Node 20?

File: docs/src/microsites/features/appearance.md [hunk 1]
```diff
@@ -84,9 +84,6 @@ The preheader region appears before the header. It shows the sitewide search if
 - Constrained content and full width background: the content is constrained to the width of the page, while the background is full width
 - Constrained content and background: both the background and content are constrained and centered
 
-![Illustration showing the three options described above.](/microsites/images/preheaderWidth.png)
```
Comment (millnut) [inline docs/src/microsites/features/appearance.md:87]: This image does not exist so I removed this line as it was causing the build to fail

**Review by markconroy — APPROVED** [review]
FAAAAANTASTIC!

Thanks so much for all this work @millnut 

---
Thanks to [Big Blue Door](https://www.bigbluedoor.net/) for sponsoring my time to work on this.

**Comment by markconroy** [comment]
One issue we have is on large screens, the search bar is not visible between 1200px and 1350px, but let's get this deployed first and we can create a follow up issue for that (unless you want to fix it before we deploy).

**Comment by millnut** [comment]
> One issue we have is on large screens, the search bar is not visible between 1200px and 1350px, but let's get this deployed first and we can create a follow up issue for that (unless you want to fix it before we deploy).

Yep let me take a look at that before merging, I'll update here once I've taken a look

**Comment by millnut** [comment]
Hi @markconroy I've taken a look at this and the best way is going to be refactoring the navbar items to dropdowns, I've refactored the navbar items into groups with enough space for us to create a `Modules` or `Ecosystem` dropdown in future to allow more detail module documentation

**Comment by markconroy** [comment]
Another approach would be to never have a desktop navigation, and always use the mobile/hamburger approach no matter how wide the screen is.

**Pull request merged**